platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/processor/AppExitAnrTraceProcessor.kt

```diff
@@ -0,0 +1,121 @@
+// capture-sdk - bitdrift's client SDK
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+package io.bitdrift.capture.reports.processor
+
+import io.bitdrift.capture.reports.AppMetrics
+import io.bitdrift.capture.reports.DeviceMetrics
+import io.bitdrift.capture.reports.ErrorDetails
+import io.bitdrift.capture.reports.FatalIssueReport
+import io.bitdrift.capture.reports.FrameDetails
+import io.bitdrift.capture.reports.FrameType
+import io.bitdrift.capture.reports.Sdk
+import io.bitdrift.capture.reports.SourceFile
+import io.bitdrift.capture.reports.ThreadDetails
+import io.bitdrift.capture.reports.processor.FatalIssueReporterProcessor.Companion.UNKNOWN_FIELD_VALUE
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
+
+/**
+ * Process an converts an ANR report from [android.app.ApplicationExitInfo]
+ * into a io.bitdrift.capture.reports.FatalIssueReport
+ */
+internal object AppExitAnrTraceProcessor {
+    private const val ANR_MAIN_THREAD_IDENTIFIED = "\"main\""
+    private const val ANR_STACKTRACE_PREFIX = "at "
+    private val ANR_STACK_TRACE_REGEX = Regex("^\\s+at\\s+(.*)\\.(.*)\\((.*):(\\d+)\\)$")
```

Comment:
could you used named capture groups here to be more explicit? e.g. `\\s+(?<className>.*)\\.(?<symbolName>.*)` etc

Reply:
also maybe it's just me but these look quite naive matchers. I can see from your own example below that we wouldn't properly include the Nativemethod) part here? https://github.com/bitdriftlabs/capture-sdk/pull/300/files#diff-8628c740ec3777bb2eb72bfa2c44310a6c9d6e48e5d760342b30f4ccf5f96615R226

like discussed offline I would feel better if we use android's reference regexes implementations found in https://cs.android.com/android/platform/superproject/+/master:development/tools/bugreport/src/com/android/bugreport/stacks/ThreadSnapshotParser.java;l=54-57 even in these earlier explorations